lib/widgets/action_sheet.dart

```diff
@@ -902,6 +906,27 @@ class MarkAsUnreadButton extends MessageActionSheetMenuItemButton {
   }
 }
 
+class UnrevealMutedMessageButton extends MessageActionSheetMenuItemButton {
+  UnrevealMutedMessageButton({
+    super.key,
+    required super.message,
+    required super.pageContext,
+  });
+
+  @override
+  IconData get icon => ZulipIcons.eye_off;
+
+  @override
+  String label(ZulipLocalizations zulipLocalizations) {
+    return zulipLocalizations.actionSheetOptionHideMutedMessage;
+  }
+
+  @override
+  void onPressed() {
+    findMessageListPage().unrevealMutedMessage(message.id);
```

Comment:
And this could gain an assertion:

```suggestion
  void onPressed() {
    // The message should have been revealed in order to reach this action sheet.
    assert(MessageListPage.revealedMutedMessagesOf(pageContext)
      .isMutedMessageRevealed(message.id));
    findMessageListPage().unrevealMutedMessage(message.id);
```